Gordon360/Controllers/RequestsController.cs

```diff
@@ -134,56 +129,54 @@ public ActionResult<REQUEST> Put(int id, REQUEST membershipRequest)
             return Ok(membershipRequest);
         }
 
+
         /// <summary>
         /// Sets a membership request to Approved
         /// </summary>
-        /// <param name="id">The id of the membership request in question.</param>
-        /// <returns>If successful: THe updated membership request wrapped in an OK Http status code.</returns>
+        /// <param name="membershipRequestID">The id of the membership request in question.</param>
+        /// <param name="status">The status that the membership requst will be changed to.</param>
+        /// <returns>The updated request</returns>
         [HttpPost]
-        [Route("{id}/approve")]
-        [StateYourBusiness(operation = Operation.DENY_ALLOW, resource = Resource.MEMBERSHIP_REQUEST)]
-        public ActionResult<MEMBERSHIP> ApproveRequest(int id)
+        [Route("{membershipRequestID}/status")]
+        [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.MEMBERSHIP_REQUEST)]
+        public async Task<ActionResult<RequestView>> UpdateStatusAsync(int membershipRequestID, [FromBody] string status)
         {
-            var result = _membershipRequestService.ApproveRequest(id);
+            RequestView? updated = null;
 
-            if (result == null)
+            switch (status)
             {
-                return NotFound();
+                case Request_Status.APPROVED:
+                    updated = await _membershipRequestService.ApproveAsync(membershipRequestID);
+                    break;
+                case Request_Status.DENIED:
+                    updated = await _membershipRequestService.DenyAsync(membershipRequestID);
+                    break;
+                case Request_Status.PENDING:
+                    updated = await _membershipRequestService.SetPendingAsync(membershipRequestID);
+                    break;
+                default:
+                    return BadRequest($"Status must be one of '{Request_Status.APPROVED}', '{Request_Status.DENIED}', or '{Request_Status.PENDING}'");
             }
```

Comment:
A switch expression would be simpler here:
```suggestion
RequestView? updated = status switch
            {
                Request_Status.APPROVED => await _membershipRequestService.ApproveAsync(membershipRequestID),
               Request_Status.DENIED => await _membershipRequestService.DenyAsync(membershipRequestID),
                Request_Status.PENDING => await _membershipRequestService.SetPendingAsync(membershipRequestID),
                _ => throw new BadInputException($"Status must be one of '{Request_Status.APPROVED}', '{Request_Status.DENIED}', or '{Request_Status.PENDING}'");
            }
```